pyalex/api.py

```diff
@@ -321,6 +331,11 @@ def select(self, s):
         self._add_params("select", s)
         return self
 
+    def autocomplete(self, s, **kwargs):
+        """ autocomplete the string s, for a specific type of entity """
+        self._add_params("q", s)
+        return self.get(**kwargs)
```

Comment:
My suggestion would be to implement it like this:

```suggestion
    def autocomplete(self, q, return_meta=False):
        """Autocomplete query q for entity"""

        self._add_params("q", q)

        # manipulate URL for autocomplete
        url_split = urlsplit(self.url)
        new_url = urlunsplit(
            (
                url_split.scheme,
                url_split.netloc,
                f"/autocomplete{url_split.path}",
                url_split.query,
                url_split.fragment,
            )
        )

        return self._get_from_url(
            new_url, return_meta=return_meta, resource_class=Autocomplete
        )
```

Also, it's not the nicest option, but it keeps the autocomplete functionality separated from everything else. We don't need the changes above. As I don't think this was the most intuitive implementation OpenAlex chose, we might benefit from keeping it separate from the rest of the class. 

I changed `_get_from_url` to accept an extra argument.